crates/fs/private/hamt/hash.rs

```diff
@@ -16,11 +16,38 @@ const MAX_CURSOR_DEPTH: usize = HASH_BYTE_SIZE * 2;
 //--------------------------------------------------------------------------------------------------
 
 /// A common trait for the ability to generate a hash of some data.
+///
+/// # Examples
+///
+/// ```
+/// use sha3::{Digest, Sha3_256};
+/// use wnfs::{Hasher, HashOutput};
+///
+/// struct MyHasher;
+///
+/// impl Hasher for MyHasher {
+///     fn hash<D: AsRef<[u8]>>(data: &D) -> HashOutput {
+///         let mut hasher = Sha3_256::new();
+///         hasher.update(data.as_ref());
+///         hasher.finalize().into()
+///     }
+/// }
+/// ```
 pub trait Hasher {
     /// Generates a hash of the given data.
-    fn hash<K: AsRef<[u8]>>(key: &K) -> HashOutput;
+    fn hash<D: AsRef<[u8]>>(data: &D) -> HashOutput;
 }
 
+/// HashNibbles is a wrapper around a byte slice that provides a cursor for traversing the nibbles.
+///
+/// # Examples
+///
+/// ```
+/// use wnfs::private::HashNibbles;
+///
+/// let nibbles = HashNibbles::new(&[0xFFu8; 32]);
+/// println!("{:?}", nibbles);
+/// ```
 #[derive(Debug, Clone)]
 pub struct HashNibbles<'a> {
```

Comment:
Does this actually need to be a `pub` thing?
I guess now it's documented and we can keep it at that, but just wondering whether we want to export the guts of the HAMT.